### ISSUE TITLE
.npmrc containing undefined environment variables gets completely ignored since pnpm v7.20.0

### ISSUE THREAD

**Issue by Kage0x3B** [pnpm/pnpm#5914]
### Code to reproduce the issue:
`.npmrc`:
```
@my-company:registry=https://npm.pkg.github.com
//npm.pkg.github.com/:_authToken=${NOT_EXISTING_ENV_VARIABLE}
```

Execute `pnpm config list`.

### Expected behavior:

Before v7.20 with the config changes, pnpm issued a warning (`WARN Issue while reading "/home/.../my-project/.npmrc". Failed to replace env in config: ${NOT_EXISTING_ENV_VARIABLE}`) but still read the other config entries.
We were using this to set the registry authToken in a docker container by setting the env variable but defaulting to using the github npm registry via the normal `pnpm login` while developing locally.

### Actual behavior:

Pnpm issues the warning about the missing env variable but also ignores the whole config file.

### Additional information:

 - `node -v` prints: v16.19.0
 - Windows, macOS, or Linux?: Linux

**Comment by Kage0x3B** [comment]
I tried looking into the cause and found the error message in [env-replace.ts](https://github.com/pnpm/components/blob/f2f248e3ab5006f87cba492a892733286a236bfc/config/env-replace/env-replace.ts#L13) but I'm currently confused, why the error message is exactly ` WARN  Issue while reading "/home/.../my-project/.npmrc". Failed to replace env in config: ${NOT_EXISTING_ENV_VARIABLE}`. `envReplace` seems to be used only once in [readLocalConfig.ts](https://github.com/pnpm/pnpm/blob/841f52e709c301a8eb738573de549eea2b13f978/config/config/src/readLocalConfig.ts#L19) where the exception is immediately caught and ignored, which seems right. But the error message still shows up and all other entries in the config files aren't loaded so I'm missing something.

**Comment by pelletier197** [comment]
Got the same issue on my side. Have you managed to find a workaround? This stops me from using pnpm completely on my project at the moment. This is kind of a severe issue.

**Comment by Kage0x3B** [comment]
It's not really a workaround, I've switching to using my Github Packages Api Key via environment variables for docker and during development by adding the api key environment variable to my `.bashrc`. Besides that, downgrading to `v7.19` might be your only option.

**Comment by pelletier197** [comment]
Okay, on my side I figured out the issue. It was a dumb attention error :sweat_smile: 

Rather than doing 
```
export NPM_TOKEN="xxx" 
```
inside my `,bashrc`, I was doing only 
```
NPM_TOKEN="xxx"
```
Therefore, the variable was not exported properly and not accessible in my environment variables. 

It was weird because I was able to do `echo $NPM_TOKEN` and see it, but I could not see it when doing `env | grep NPM_TOKEN`.

**Comment by a11rew** [comment]
Bumping this, this is a problem because dependencies are not installed correctly in workspaces (according to the config in the npmrc) once parsing fails. 

The undefined environment variable issue is already output as a warning, the rest of the npmrc should be parsed instead of ignored

**Comment by fk-aconso** [comment]
> The undefined environment variable issue is already output as a warning, the rest of the npmrc should be parsed instead of ignored

Run into this issue and agree here.

**Comment by Jak-Ch-ll** [comment]
As a workaround you can fallback the variables to anything:
```
//npm.pkg.github.com/:_authToken=${NOT_EXISTING_ENV_VARIABLE:-""}
```

The downside (or additional upside): It suppresses the warning

**Comment by tomas223** [comment]
@Jak-Ch-ll I stumbled on exactly the same issue.

Variable FALLBACK in `.npmrc` works normally with the exception of `pnpm publish`.

This should probably be a separate issue.

(and thanks for the workaround)

**Comment by zkochan** [comment]
This no longer reproduces on current pnpm, so closing.

Verified on pnpm 12.5.1: an .npmrc whose auth line references an unset environment variable is still read. `pnpm config list` shows the other entries from the same file.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
